clang/lib/Driver/Driver.cpp

```diff
@@ -801,14 +816,19 @@ void Driver::CreateOffloadingDeviceToolChains(Compilation &C,
           << SYCLTargetsValues->getAsString(C.getInputArgs());
   } else {
     // If -fsycl is supplied without -fsycl-targets we will assume SPIR-V
+    // unless -fintelfpga is supplied, which uses SPIR-V with fpga AOT.
     if (HasValidSYCLRuntime) {
       const ToolChain *HostTC =
           C.getSingleOffloadToolChain<Action::OFK_Host>();
       llvm::Triple TT(TargetTriple);
-      TT.setArch(llvm::Triple::spir64);
-      TT.setVendor(llvm::Triple::UnknownVendor);
-      TT.setOS(llvm::Triple(llvm::sys::getProcessTriple()).getOS());
-      TT.setEnvironment(llvm::Triple::SYCLDevice);
+      if (SYCLfpga)
+        TT.setTriple("spir64_fpga-unknown-linux-sycldevice");
```

Comment:
this probably won't work for Windows. I was thinking about using the triple components provided in -fsycl-targets, including OS, rather then overriding it here. If no OS was provided, then llvm::Triple(llvm::sys::getProcessTriple()).getOS()) seems like the proper choice. Does this sound reasonable?

Reply:
The OS component of target triple should be unknown, as devices do not have any OS (as well as environment/runtime).
This information should be inferred from the aux-triple, which defines the host OS and environment.

I suggest fixing this in a separate patch, as it might require changes in other components also, not just the driver.

Reply:
I will update the default triple for -fintelfpga to match what Konst has suggested.  We can make adjustments using 'unknown' for the OS later as Alexey stated.